### PR TITLE
option to truncate dn

### PR DESCRIPTION
**Pull request by GOID1989**

### CODE REVIEW

File: lib/zabbixldapconf.py [hunk 1]
```diff
@@ -73,6 +73,7 @@ def __init__(self, config: str):
                                                   fallback='(&(objectClass=posixAccount)(uid=%s))', raw=True)
             self.openldap_groupattribute = parser.get('openldap', 'groupattribute', fallback='memberUid', raw=True)
             self.openldap_userattribute = parser.get('openldap', 'userattribute', fallback='uid', raw=True)
+            self.openldap_truncatedn = parser.get('openldap', 'truncatedn', fallback='false', raw=True)
```
Comment (scoopex) [inline lib/zabbixldapconf.py]: Can you use try_get_item_bool to parse the value?

File: zabbix-ldap.conf.example [hunk 2]
```diff
@@ -25,6 +25,7 @@ filtergroup = (&(objectClass=posixGroup)(cn=%s))
 filteruser = (&(objectClass=posixAccount)(uid=%s))
 groupattribute = memberUid
 userattribute = uid
+truncatedn = false
```
Comment (scoopex) [inline zabbix-ldap.conf.example]: Can you add this to the Readme instead?
(Including why this is needed and a n example of the truncation)?

**Comment by scoopex** [comment]
Welcome to the circle of contributors :-)
I added some comments to your PR.

Can you combine  your changes in a single and descriptive commit message?
(see also https://jvns.ca/blog/2023/11/06/rebasing-what-can-go-wrong-/ .... i.e. by `git rebase -i HEAD^^`, add "s" to all your commits except the first one and `git push --force` is your friend)

**Review by scoopex — COMMENTED** [review]
Welcome to the circle of contributors :-)
I added some comments to your PR.

Can you combine your changes in a single and descriptive commit message?
(see also https://jvns.ca/blog/2023/11/06/rebasing-what-can-go-wrong-/ .... i.e. by git rebase -i HEAD^^, add "s" to all your commits except the first one and git push --force is your friend)

**Comment by GOID1989** [comment]
check it

File: README.md [hunk 3]
```diff
@@ -90,6 +90,7 @@ In order to use the *zabbix-ldap-sync* script we need to create a configuration
 * `filteruser` = The ldap filter to get the users in OpenLDAP mode, by default `(&(objectClass=posixAccount)(uid=%s))`
 * `groupattribute` = The attribute used for membership in a group in OpenLDAP mode, by default `memberUid`
 * `userattribute` = The attribute for users in openldap mode, by default `uid`
+* `truncatedn` - If set to true distinguished name (DN) will be truncated, by default false. Group members usually defined as `full-path-DN` not clear `login` and it's breaks search. Example: `uid=testuser,cn=users,cn=accounts,dc=example,dc=com` cut to `uid=testuser`
```
Comment (scoopex) [inline README.md]: Thanks :-)

Sorry about the nitpick :-) Lets make this to one additional level more clearer.

```
`truncatedn` - If set to true the distinguished name (DN) will be truncated to the first component (by default false). Group members are usually defined as `full-path-DN`. If your ldap server just uses the `login` names to reference group members (i.e. FreeIPA) you can use this functionality to solve problems with broken searches. Example: `uid=testuser,cn=users,cn=accounts,dc=example,dc=com` cut to `uid=testuser`
```

Comment (GOID1989) [reply README.md]: sorry, for bad english

**Comment by scoopex** [comment]
Many thanks!!!!

**Pull request merged**